### PR TITLE
bump actions per dependabot

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

Fixes pyrsia/pyrsia#572 which includes dependabot PRs #581 #575 #576 #566 #544

This PR bumps all of the actions to the recommended version by dependabot.  The new action versions work as expected in the lint.yaml and rust.yaml.

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [X] I've requested a review  from `pyrsia/collaborators`.

### Code Contributions

- [X] I've built the code `cargo build --all-targets` successfully.
- [X] I've run the unit tests `cargo test --workspace` and everything passes.

### CODE REVIEW

**Comment by sbtaylor15** [comment]
The dependabot PRs should be closed automatically when this PR is merged.  I will keep an eye on them to make sure they are closed correctly.

**Review by tiainen — DISMISSED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     - uses: actions/checkout@v3
   
     # Use sscache store in GitHub cache
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3.0.2
```
Comment (prince-chrismc) [inline .github/workflows/rust.yml:22]:
```suggestion
    - uses: actions/cache@v3
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
     - uses: actions/checkout@v3
 
     # Use sscache store in GitHub cache
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3.0.2
```
Comment (prince-chrismc) [inline .github/workflows/rust.yml:56]:
```suggestion
    - uses: actions/cache@v3
```

**Comment by prince-chrismc** [comment]
Please use the major tags so we dont have to do this as often 🙏

File: .github/workflows/rust.yml (hunk 2 above)
Comment (sbtaylor15) [reply .github/workflows/rust.yml:56]: Leaving as 3.0.2

image

File: .github/workflows/rust.yml (hunk 1 above)
Comment (sbtaylor15) [reply .github/workflows/rust.yml:22]: Leaving as 3.0.2

image

**Review by prince-chrismc — APPROVED** [review]
LGTM!!! 🚀

**Review by fishseabowl — APPROVED** [review]
LGTM

**Pull request merged**